<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from sda10f.xacro                   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="motoman_sda10f" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_base.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_base.stl"/>
      </geometry>
      <material name="yellow">
        <color rgba="0 1 1 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.00094542"/>
      <inertia ixx="2e-08" ixy="0" ixz="0" iyy="3e-08" iyz="0" izz="2e-08"/>
    </inertial>
  </link>
  <link name="torso_link_b1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_b1.stl"/>
      </geometry>
      <material name="silver">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_b1.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="torso_link_b2" />
  <joint name="torso_joint_b1" type="revolute">
    <parent link="base_link"/>
    <child link="torso_link_b1"/>
    <origin rpy="0 0 0" xyz="0.09257 0 0.8835"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.9570" upper="2.9570" velocity="2.26"/>
  </joint>
  <joint name="torso_joint_b2" type="revolute">
    <parent link="base_link"/>
    <child link="torso_link_b2"/>
    <origin rpy="0 0 0" xyz="0.09257 0 0.8835"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-2.9570" upper="2.9570" velocity="2.26"/>
    <mimic joint="torso_joint_b1" multiplier="1" offset="0"/>
  </joint>
  <link name="arm_left_link_1_s">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_s.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_s.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_2_l">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_l.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_l.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_3_e">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_e.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_e.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_4_u">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_u.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_u.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_5_r">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_r.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_r.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_6_b">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_b.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_b.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_7_t">
    <visual>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_t.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_t.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_left_link_tool0"/>
  <joint name="arm_left_joint_1_s" type="revolute">
    <parent link="torso_link_b1"/>
    <child link="arm_left_link_1_s"/>
    <origin rpy="1.57 0 0.0" xyz="0.09996 0.0275 0.32214"/>
    <axis xyz="0 0 -1.0"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="2.95"/>
  </joint>
  <joint name="arm_left_joint_2_l" type="revolute">
    <parent link="arm_left_link_1_s"/>
    <child link="arm_left_link_2_l"/>
    <origin rpy="-1.57 0 0" xyz="0 0.0577 -0.2255"/>
    <axis xyz="0 0 -1.0"/>
    <limit effort="0" lower="-1.90" upper="1.90" velocity="2.95"/>
  </joint>
  <joint name="arm_left_joint_3_e" type="revolute">
    <parent link="arm_left_link_2_l"/>
    <child link="arm_left_link_3_e"/>
    <origin rpy="1.57 0 0" xyz="0 0.1690 -0.0454"/>
    <axis xyz="0 0 -1.0"/>
    <limit effort="0" lower="-2.95" upper="2.95" velocity="2.95"/>
  </joint>
  <joint name="arm_left_joint_4_u" type="revolute">
    <parent link="arm_left_link_3_e"/>
    <child link="arm_left_link_4_u"/>
    <origin rpy="-1.57 0 0" xyz="0 -0.0085 -0.181"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-2.36" upper="2.36" velocity="2.95"/>
  </joint>
  <joint name="arm_left_joint_5_r" type="revolute">
    <parent link="arm_left_link_4_u"/>
    <child link="arm_left_link_5_r"/>
    <origin rpy="1.57 0 0" xyz="0 0.1936 0.0155"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="3.48"/>
  </joint>
  <joint name="arm_left_joint_6_b" type="revolute">
    <parent link="arm_left_link_5_r"/>
    <child link="arm_left_link_6_b"/>
    <origin rpy="-1.57 0 0" xyz="0 0.0080 -0.155"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-1.90" upper="1.90" velocity="3.48"/>
  </joint>
  <joint name="arm_left_joint_7_t" type="revolute">
    <parent link="arm_left_link_6_b"/>
    <child link="arm_left_link_7_t"/>
    <origin rpy="1.57 0 0" xyz="0 0.168 -0.015"/>
    <axis xyz="0 0 -1.0"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="6.97"/>
  </joint>
  <joint name="arm_left_joint_tool0" type="fixed">
    <origin rpy="0 0 -3.13" xyz="0 0 -0.05"/>
    <parent link="arm_left_link_7_t"/>
    <child link="arm_left_link_tool0"/>
  </joint>
  <link name="arm_right_link_1_s">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_s.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0.8 0.8 0.8 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_s.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_2_l">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_l.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_l.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_3_e">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_e.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_e.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_4_u">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_u.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_u.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_5_r">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_r.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_r.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_6_b">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_b.stl"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_b.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_7_t">
    <visual>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/visual/motoman_axis_t.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://prx_simulation/urdf/meshes/sda10f/collision/motoman_axis_t.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="arm_right_link_tool0"/>
  <joint name="arm_right_joint_1_s" type="revolute">
    <parent link="torso_link_b1"/>
    <child link="arm_right_link_1_s"/>
    <origin rpy="1.57 0 -3.14" xyz="0.09996 -0.0275 0.32214"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="2.95"/>    
  </joint>
  <joint name="arm_right_joint_2_l" type="revolute">
    <parent link="arm_right_link_1_s"/>
    <child link="arm_right_link_2_l"/>
    <origin rpy="-1.57 0 0" xyz="0 0.0577 -0.2255"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="0" lower="-1.90" upper="1.90" velocity="2.95"/>
  </joint>
  <joint name="arm_right_joint_3_e" type="revolute">
    <parent link="arm_right_link_2_l"/>
    <child link="arm_right_link_3_e"/>
    <origin rpy="1.57 0 0" xyz="0 0.1690 -0.0454"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="0" lower="-2.95" upper="2.95" velocity="2.95"/>
  </joint>
  <joint name="arm_right_joint_4_u" type="revolute">
    <parent link="arm_right_link_3_e"/>
    <child link="arm_right_link_4_u"/>
    <origin rpy="-1.57 0 0" xyz="0 -0.0085 -0.181"/>
    <axis xyz="0 0 -1"/>
    <limit effort="0" lower="-2.36" upper="2.36" velocity="2.95"/>
  </joint>
  <joint name="arm_right_joint_5_r" type="revolute">
    <parent link="arm_right_link_4_u"/>
    <child link="arm_right_link_5_r"/>
    <origin rpy="1.57 0 0" xyz="0 0.1936 0.0155"/>
    <axis xyz="0 0 -1"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="3.48"/>
  </joint>
  <joint name="arm_right_joint_6_b" type="revolute">
    <parent link="arm_right_link_5_r"/>
    <child link="arm_right_link_6_b"/>
    <origin rpy="-1.57 0 0" xyz="0 0.0080 -0.155"/>
    <axis xyz="0 0 -1"/>
    <limit effort="0" lower="-1.90" upper="1.90" velocity="3.48"/>
  </joint>
  <joint name="arm_right_joint_7_t" type="revolute">
    <parent link="arm_right_link_6_b"/>
    <child link="arm_right_link_7_t"/>
    <origin rpy="1.57 0 0" xyz="0 0.168 -0.015"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="0" lower="-3.13" upper="3.13" velocity="6.97"/>
  </joint>
  <joint name="arm_right_joint_tool0" type="fixed">
    <origin rpy="0 0 -3.13" xyz="0 0 0.0"/>
    <parent link="arm_right_link_7_t"/>
    <child link="arm_right_link_tool0"/>
  </joint>

  <!-- Gripper -->

  <link name="base_plate">
    <visual>
      <geometry>
        <cylinder radius="0.04" length="0.008" />
      </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <cylinder radius="0.04" length="0.008" />
        </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="main_beam">
    <visual>
      <geometry>
        <box size="0.015 0.040 0.180" />
      </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.0" rpy="0 0 0" />
        <geometry>
          <box size="0.015 0.040 0.180" />
        </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="actuator">
    <visual>
      <geometry>
        <box size="0.03 0.035 0.180" />
      </geometry>
      <material name="dark_grey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.0" rpy="0 0 0" />
      <geometry>
        <box size="0.03 0.035 0.180" />
      </geometry>
      <material name="dark_grey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="head_connector">
    <visual>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
      <geometry>
        <box size="0.035 0.015 0.039" />
      </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
        <geometry>
          <box size="0.035 0.015 0.039" />
        </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="head_plate">
    <visual>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
      <geometry>
        <box size="0.045 0.045 0.005" />
      </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
        <geometry>
          <box size="0.045 0.045 0.005" />
        </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="head_base">
    <visual>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
      <geometry>
        <box size="0.062 0.121 0.03" />
      </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
        <geometry>
          <box size="0.062 0.121 0.03" />
        </geometry>
      <material name="silver">
        <color rgba="0.76 0.76 0.82 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="head_sponge">
    <visual>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
      <geometry>
        <box size="0.06 0.12 0.016" />
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
        <geometry>
          <box size="0.06 0.12 0.016" />
        </geometry>
      <material name="black">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  
  <link name="tube_port">
    <visual>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.01" length="0.025" />
      </geometry>
      <material name="blue">
        <color rgba="0.1 0.1 0.92 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="-0.01 0.0 -0.004" rpy="0 0 0" />
        <geometry>
          <cylinder radius="0.01" length="0.025" />
        </geometry>
      <material name="blue">
        <color rgba="0.1 0.1 0.92 1"/>
      </material>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  

<!-- Joints -->

  <joint name="plate_beam" type="fixed">
    <origin xyz="0.0 0.0 0.094" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="base_plate" />
    <child link="main_beam" />
  </joint>

  <joint name="beam_act" type="fixed">
    <origin xyz="-0.0225 0.0 0.0" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="main_beam" />
    <child link="actuator" />
  </joint>

  <joint name="neck_plate" type="fixed">
    <origin xyz="0.0 0 0.022" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="head_connector" />
    <child link="head_plate" />
  </joint>

  <joint name="plate_head" type="fixed">
    <origin xyz="0.0 0 0.0175" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="head_plate" />
    <child link="head_base" />
  </joint>

  <joint name="head_sponge" type="fixed">
    <origin xyz="0.0 0 0.023" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="head_base" />
    <child link="head_sponge" />
  </joint>

  <joint name="head_valve" type="fixed">
    <origin xyz="0.0 0.0435 -0.0275" rpy="0 0 0" />
    <axis xyz="0 0 0" />
    <parent link="head_base" />
    <child link="tube_port" />
  </joint>

  <joint name="head_hinge" type="revolute">
    <origin xyz="0.0 0.0 0.0825" rpy="0 0 0" />
    <axis xyz="0 1 0" />
    <parent link="main_beam" />
    <child link="head_connector" />
    <dynamics damping="0.0" friction="0.5" />
    <limit effort="50" velocity="10" lower="-0.05" upper="1.6" />
  </joint>


  <!-- Then, we need to attach the gripper to the robot's arm -->

  <joint name="left_tool_base_joint" type="fixed">
    <origin rpy="3.1416 0 3.1416" xyz="0 0 0"/>
    <parent link="arm_left_link_tool0"/>
    <child link="base_plate"/>
  </joint>

  <xacro:motoman_sda10f prefix=""/>
</robot>

